### ISSUE TITLE
Size limits behaviour differs in `--replay` mode

### ISSUE THREAD

**Issue by HermannDppes** [sahib/rmlint#368]
When researching #367, I found out about the following behaviour which I found odd. I had expected a size parameter to matter only if the size is not implicitly zero anyway in which case the additional specification of a size makes no sense. Further, this is – according to my understanding – the behaviour outlined in the manpage (emphasis mine):
> Only consider files **as duplicates** in a certain size range.

# Actual Behaviour

When passing the arguments `--size=1024-2048 -T 'ed,dd'`, rmlint searches for files having one of the following two characteristics:
* The file is empty and has a size between one and two kibibyte.
* The file is duplicated and has a size between one and two kibibyte.

# Expected Behaviour

When passing the arguments `--size=1024-2048 -T 'ed,dd'`, rmlint searches for files having one of the following two characteristics:
* The file is empty.
* The file is duplicated and has a size between one and two kibibyte.

# Related Behaviour

The same happens for empty directories and duplicated directories which both seem to default to a size of zero. (Which makes some sense for the former but none for the latter if the application of the size limits in this case is intentional.)

# Limitations

I only checked the code for replaying previous runs. Therefore, it might be a bug specific to this.

**Comment by HermannDppes** [comment]
Upon further investigation and thought, there seem to be (at least) two bugs in the `--replay` mode:
* Size limits are applied to everything, not only to duplicate files
* The size of duplicate directories is neither correctly determined (always `0`) nor used to calculate the total size of lint found.

For the latter, I hacked the following workaround. It doesn't feel right but seems to work. (Did I even write the correct field?)

```udiff
diff --git a/lib/replay.c b/lib/replay.c
index fad16294..c7583593 100644
--- a/lib/replay.c
+++ b/lib/replay.c
@@ -190,6 +190,10 @@ static RmFile *rm_parrot_try_next(RmParrot *polly) {
     file->digest = rm_digest_new(RM_DIGEST_EXT, 0);
     file->free_digest = true;
 
+    if (type == RM_LINT_TYPE_DUPE_DIR_CANDIDATE) {
+        file->actual_file_size = json_object_get_int_member(object, "size");
+    }
+
     if(file->is_original) {
         polly->last_original = file;
     }
@@ -412,6 +416,15 @@ static void rm_parrot_update_stats(RmParrotCage *cage, RmFile *file) {
                 session->total_lint_size += file->actual_file_size;
             }
         }
+    } else if (file->lint_type == RM_LINT_TYPE_DUPE_DIR_CANDIDATE) {
+        session->dup_group_counter += file->is_original;
+        if(!file->is_original) {
+            session->dup_counter += 1;
+
+            if(!RM_FILE_IS_HARDLINK(file)) {
+                session->total_lint_size += file->actual_file_size;
+            }
+        }
     } else {
         session->other_lint_cnt += 1;
     }
```

**Comment by SeeSpotRun** [comment]
If this wasn't already resolved, I think this is resolved by #486.

Note: `--size`= applies to file sizes.  So `--size=1024-2048 -T 'ed,dd'` will find duplicate dir's in which *all files* are either zero length or between 1024 and 2048 bytes long.

Closing now.
